### ISSUE TITLE
Tearsheet: invalid aria-label

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#2978]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

The `Tearsheet#title` property is a ReactNode:

```
  /**
   * The main title of the tearsheet, displayed in the header area.
   */
  title: PropTypes.node
```

But TearsheetShell.js assigns it to `aria-label`.

So when you set `title` to something like `<div>...</div>` it ends up with the obviously invalid:

```
<div aria-label="[object Object]">
```

(It's invalid because the label needs to be human-readable text that describes the dialog.)

There should be some way to specify the `aria-label` or (better yet) `aria-labelledby` for the dialog.

**Comment by elycheea** [comment]
Looks like we’re overriding the arial labels from ComposedModal. We can look into this!
